test/unit_tests/sunlinsol/ginkgo/test_sunlinsol_ginkgo_block.cpp

```diff
@@ -0,0 +1,433 @@
+    }
+  }
+}
+
+/* -------------------------------------------------------------------------- *
+ * Global Executor for sync_device                                            *
+ * -------------------------------------------------------------------------- */
+// sycl only provides synchronize on queue
+std::shared_ptr<const gko::Executor> global_exec;
+
+/* ----------------------------------------------------------------------
+  * SUNLinSol_Ginkgo Testing Routine
+  * --------------------------------------------------------------------*/
+int main(int argc, char* argv[])
+{
+  int argi{0};
+  int fails{0}; /* counter for test failures */
+
+  sundials::Context sunctx;
+
+#if defined(USE_HIP)
+  auto gko_exec{gko::HipExecutor::create(0, gko::OmpExecutor::create())};
+#elif defined(USE_CUDA)
+  auto gko_exec{gko::CudaExecutor::create(0, gko::OmpExecutor::create())};
+#elif defined(USE_SYCL)
+  auto gko_exec{gko::DpcppExecutor::create(0, gko::ReferenceExecutor::create())};
+#elif defined(USE_OMP)
+  auto gko_exec{gko::OmpExecutor::create()};
+#else
+  auto gko_exec{gko::ReferenceExecutor::create()};
+#endif
+
+  // For sync_device
+  global_exec = gko_exec;
+
+  /* ------------ *
+   * Check inputs *
+   * ------------ */
+
+  if (argc < 8)
+  {
+    std::cerr << "ERROR: SEVEN (7) inputs required:\n"
+              << "  1) batch method\n"
+              << "  2) matrix type (csr or dense)\n"
+              << "  3) number of matrix columns in a block\n"
+              << "  4) number of matrix blocks"
+              << "  5) condition number of a block\n"
+              << "  6) max iterations per block\n"
+              << "  7) print timing\n";
+    return 1;
+  }
+
+  std::string method{argv[++argi]};
+  std::transform(method.begin(), method.end(), method.begin(),
+                 [](unsigned char c) { return std::tolower(c); });
+
+  if (!methods.count(method))
+  {
+    std::cerr << "ERROR: method must be one of ";
+    for (const auto& m : methods) { std::cout << m.first << ", "; }
+    std::cout << std::endl;
+    return 1;
+  }
+
+  std::string matrix_type{argv[++argi]};
+  std::transform(matrix_type.begin(), matrix_type.end(), matrix_type.begin(),
+                 [](unsigned char c) { return std::tolower(c); });
+
+  if (!matrix_types.count(matrix_type))
+  {
+    std::cerr << "ERROR: matrix type must be one of ";
+    for (const auto& m : matrix_types) { std::cout << m.first << ", "; }
+    std::cout << std::endl;
+    return 1;
+  }
+
+  const auto matcols{static_cast<sunindextype>(atoll(argv[++argi]))};
+  if (matcols <= 0)
+  {
+    std::cerr << "ERROR: number of matrix columns must be a positive integer\n";
+    return 1;
+  }
+  const auto matrows{matcols};
+
+  const auto num_blocks{static_cast<sunindextype>(atoll(argv[++argi]))};
+  if (num_blocks <= 0)
+  {
+    std::cerr << "ERROR: number of blocks must be a positive integer\n";
+    return 1;
+  }
+
+  const auto matcond{static_cast<sunrealtype>(atof(argv[++argi]))};
```

Comment:
Possibly more conventional?
```suggestion
  const auto matcond{SUNStrToReal(argv[++argi].c_str())};
```